test-js/unit/dojo/pentaho/DojoNumberSpec.js

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+define( ["dojo/number"], function( DojoNumber ) {
+
+  describe( "Dojo i18n tests", function() {
+
+    it( "Number.parse(..) should accept default locale", function() {
+      expect( DojoNumber.parse( "1234", {} ) ).toEqual( 1234 );
+    } );
+
+    it( "Dojo Number.parse(..) should accept correct locale", function() {
+      expect( DojoNumber.parse( "1234", {"locale" : "en-us"} ) ).toEqual( 1234 );
+    } );
+    
+    //Warning!
+    // This test is failing.
+    // See BISERVER-13154
+    it( "Dojo Number.parse(..) should safely accept even incorrect locale", function() {
+      expect( DojoNumber.parse( "1234", {"locale" : "en_us"} ) ).toEqual( 1234 );
```

Comment:
Given that this test will be failing (and we expect it to), can you change the assert to "expect(..).toThrow()".

This way you both state that this code is expected to fail with the current Dojo code, and your test no longer reports an failing unit-test and thus does not kill the strict build process (the ones that actually fail on test errors).

It might be also wise to update the comment to say: "This test is failing as dojo does not correctly resolve resource-bundles." This simple note saves later developers from having to dig though an ancient JIRA case just to understand why this test is here.